updating content inheritance on page builder

aachavarria opened the pull request:
### Ticket reference or full description of what's in the PR

aachavarria committed:
creating getContentInstanceByPath();
reading sourceMap ContentInstances;

aachavarria committed:
PreviewConcierge.tsx: Mapping SourceMap contentInstances;
ContentController.ts: Creating getContentInstanceByPath();
ICERegistry.ts: updating getReferentialEntries to look on sourceMap if field not found in the contentType;
GuestProxy.tsx: Fix on GuestProxy to wait until paths$;

aachavarria committed:
Resetting requestedSourceMapPaths on Refresh();

rart commented on ui/app/src/models/ContentInstance.ts:
```diff
@@ -22,7 +22,7 @@ export interface ContentInstanceSystemProps {
   dateModified: string;
   contentTypeId: string;
-  sourceMap?: { [path: string]: string }; // path: contentTypeId
+  sourceMap?: { [fieldId: string]: string }; // fieldId: path
```
I’m thinking we could have both things here. When parsing content the sourceMap would be populated with both { fieldId: path, path: contentTypeId }.

rart commented on ui/guest/src/classes/ICERegistry.ts:
```diff
@@ -324,10 +324,18 @@ export function runValidation(iceId: number, validationId: ValidationKeys, args?
+    model = contentController.getContentInstanceByPath(model.craftercms.sourceMap[record.fieldId]);
+    contentTypeId = Model.getContentTypeId(model);
+    contentType = contentController.getCachedContentType(contentTypeId);
+    field = record.fieldId ? contentTypeUtils.getField(contentType, record.fieldId) : null;
```
Add `inherited: boolean`?

rart commented on ui/guest/src/controls/rte.ts:
```diff
@@ -113,7 +114,8 @@ export function initTinyMCE(
           const content = getContent();
           if (changed) {
-            contentController.updateField(record.modelId, field.id, record.index, content);
+            const model = getReferentialEntries({ ...record, fieldId: field.id }).model;
```
Why not just sent the record? The records field and the field should match.

aachavarria replied:
rollback

rart commented on ui/guest/src/react/GuestProxy.tsx:
```diff
@@ -372,7 +373,8 @@ export default function GuestProxy() {
           const model = getCachedModel(modelId);
           const contentType = getCachedContentType(model.craftercms.contentTypeId);
-          const fieldType = ContentType.getField(contentType, fieldId).type;
+          const field = ContentType.getField(contentType, fieldId);
```
Not using field later so looks like you can roll this back?

aachavarria committed:
adding inherited support to ZoneMarker.tsx

aachavarria committed:
adding craftercms-zone-marker-icon class to ZoneMarker.tsx;
ICERegistry.ts validation;

aachavarria committed:
adding modelIdByPath to app Store;
Adding InheritedFields support on EditFormPanel.tsx;

aachavarria committed:
adding sourceMap validation;
adding InheritedField support to PreviewConcierge.tsx( InsertComponent and UpdateField) operations;

aachavarria committed:
adding InheritedField to INSERT_COMPONENT_OPERATION;
WIP( DELETE, SORT, MOVE)

aachavarria committed:
Adding InheritedField support to SORT_ITEM_OPERATION and DELETE_ITEM_OPERATION

aachavarria committed:
adding InheritedField support to MOVE_ITEM_OPERATION;

aachavarria committed:
adding comment on moveItem

rart commented on ui/app/src/services/content.ts:
```diff
@@ -313,7 +313,12 @@ export function moveItem(
+    (nnou(originalParentModelId) && nnou(targetParentModelId) && originalParentModelId === targetParentModelId)
+  ) {
     if (originalParentModelId === targetParentModelId && nnou(originalParentModelId)) {
       debugger;
```
👆

aachavarria replied:
removed

rart commented on ui/app/src/utils/model.ts:
```diff
@@ -102,3 +104,17 @@ function extractCollectionPiece(model, fieldId, index) {
 }
+
+export function isInheritedField(model, fieldId) {
+  return !!model.craftercms.sourceMap?.[fieldId];
```
`!!model.craftercms.sourceMap?.[fieldId]` => `Boolean(model.craftercms.sourceMap?.[fieldId])`

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/contentInheritance

aachavarria committed:
Pr Feedback
